extjsdk/README.md

```diff
@@ -0,0 +1,93 @@
+
+#### Configuration
+The configuration handler receives the message containing a configuration document created on the Vantiq client. This message is only received upon a successful source connection, and contains the source's configuration document as a Map at `((Map) <message>.getObject()).get("config")`. The configuration handler receives an ExtensionServiceMessage.
+
+#### Reconnection
+Under certain circumstances, typically a change to the source's configuration document, the Vantiq server may disconnect from the source. The reconnection handler is called when this occurs. It is suggested that in this handler you reset any changes caused by the configuration document and then reconnect using `client.connectToSource()`. If you do not want to send the reconnect message yourself or do not wish to create your own handler, then calling `client.setAutoReconnect(true)` will tell the Client to automatically try to reconnect once any handlers are finished The reconnection handler receives an ExtensionServiceMessage.
+
+#### Publish
+The Publish handler is called when a message is received that was generated by Vantiq with `PUBLISH <object> TO SOURCE <source name>`. The published object can be retrieved through `<message>.getObject()`. Note that anything declared with the `USING` keyword will also be placed into the same location, e.g. `PUBLISH {"hello":"world"} TO SOURCE <sourceName> USING {"option":"one"}` will generate an object that looks like `{"hello":"world","option":"one"}`. The Publish handler receives an ExtensionServiceMessage.
+
+#### <a name="queryHandler" id="queryHandler"></a>Query
+The Query handler is called when a message is received that was generated by Vantiq with `SELECT <keys> FROM SOURCE <source name>`. Query messages expect a response, either [data](#queryResponse) or an [error](#queryError), so every Client has a default handler is created that sends back an error stating that no Query handler was set. Options specified using the `WITH` keyword are received as a Map obtained with `<message>.getObject()`.
+
+### <a name="listener" id="listener"></a>ExtensionWebSocketListener
+The ExtensionWebSocketListener class should only be accessed and used indirectly through handlers. If you do find a reason to access it directly, you can use `ExtensionWebSocketClient.getListener()`, but all functionality interactions with a Listener should be performed through an ExtensionWebSocketClient. If you do need to save a reference to a listener, be aware that a Client's Listener is stopped and replaced with a functionally identical Listener just before the [close handler](#closeHandler) is called. Call `listener.isStopped()` to check if this has occurred.
+
+### <a name="extSvcMsg" id="extSvcMsg"></a>ExtensionServiceMessage
+The ExtensionServiceMessage class is a helper that tightly defines what can be in a message to or from a source, and has getters for the most relevant properties. 
```

Comment:
I think the `ExtensionServiceMessage` is the class that represents the messages that are sent to/from the server.  There are some helper methods to make it easier to obtain the properties of the message, but the class is, really, more than a helper class.